SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMigration/SpatialTestCharacterMigration.cpp

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestCharacterMigration.h"
+#include "Components/BoxComponent.h"
+#include "Engine/TriggerBox.h"
+#include "GameFramework/PlayerController.h"
+#include "Kismet/GameplayStatics.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/TestActors/TestMovementCharacter.h"
+
+/**
+ * This test moves a character backward and forward repeatedly between two workers, adding actors. Based on the SpatialTestCharacterMovement
+ * test. This test requires the CharacterMovementTestGameMode, trying to run this test on a different game mode will fail.
+ *
+ * The test includes two servers and one client worker. The client worker begins with a PlayerController and a TestCharacterMovement
+ *
+ */
+
+ASpatialTestCharacterMigration::ASpatialTestCharacterMigration()
+	: Super()
+{
+	Author = "Victoria";
+	Description = TEXT("Test Character Migration");
+	TimeLimit = 300;
+}
+
+void ASpatialTestCharacterMigration::OnOverlapBeginDestination(AActor* OverlappedActor, AActor* OtherActor)
+{
+	ATestMovementCharacter* OveralappingCharacter = Cast<ATestMovementCharacter>(OtherActor);
```

Comment:
Our coding standards ask to move potentially null pointer var declaration into the subsequent if statement. There's a few instances of that in this file.

Reply:
Sure, fixed 4 instances.